system/Test/FeatureTestTrait.php

```diff
@@ -302,6 +322,26 @@ protected function setupRequest(string $method, string $path = null): IncomingRe
 		return $request;
 	}
 
+	/**
+	 * Setup the custom request's headers
+	 * 
+	 * @param \CodeIgniter\HTTP\Request $request
+	 * 
+	 * @return \CodeIgniter\HTTP\IncomingRequest
+	 */
+	protected function setupHeaders(Request $request)
+	{
+		if(!empty($this->headers))
```

Comment:
This conditional can be removed if we have the default value set to `[]` instead of `null`.

Reply:
We actually need to keep this since this is a Trait and there is no guarantee that a developer has remembered to add the class property.